File: 38-count-and-say/count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string str = "1";
        for (int i = 2; i <= n; i++) {
            str = fun(str);
            // cout<<str<<endl;
        }
        return str;
    }

    string fun(string str) {
        string ans = "";
        int n = str.size();
        int i = 0, cnt = 1;

        while (i < n) {
            char ch = str[i];
            while (i + 1 < n && str[i + 1] == ch) {
                i++;
                cnt++;
            }
            ans += to_string(cnt);
            ans.push_back(ch);
            cnt = 1;
            i++;
        }
        return ans;
    }
};

File: 38-count-and-say/count-and-say.java
class Solution {
    public String countAndSay(int n) {
        StringBuilder str = new StringBuilder("1");

        for(int i=2;i<=n;i++){
            int strLen = str.length();
            int ind=0;
            StringBuilder currStr = new StringBuilder();
            while(ind<strLen){
                int cnt=1;
                char ch = str.charAt(ind);
                ind++;
                while(ind<strLen && ch == str.charAt(ind)){
                    cnt++;
                    ind++;
                }
                currStr.append(String.valueOf(cnt));
                currStr.append(String.valueOf(ch));

            }
            str = currStr;
        }
        return str.toString();
    }
}